_posts/2020-03-16-dask-usage-at-blue-yonder.markdown

```diff
@@ -1,131 +1,131 @@
-cluster can cope with workers being removed and/or new instances being added. While this is a supported use case, 
-it appeared to us that this seemed not a widely used feature. By making heavy use of it, we were able to uncover
-some bugs and to submit the corresponding fixes.
+  ---
+  layout: single
+  title:  "Dask Usage at Blue Yonder"
+  date:   2019-10-25 10:00:00 +0100
+  tags: technology python data-engineering
+  header:
+    overlay_image: assets/images/tech_gear_banner.jpg
+    overlay_filter: 0.2
+    show_overlay_excerpt: false
+  author: Andreas Merkel
+  author_profile: true
+  ---
+  # Dask Usage at Blue Yonder
+
+  Back in 2017, Blue Yonder started to look into 
+  [Dask/Distributed](https://distributed.dask.org) and how we can leverage it to power
+  our machine learning and data pipelines.
+  In 2019, we started to use it heavily in production for providing demand predicions,
+  recommended order quantities, and price proposals to our customers.
+  Time to take a a look at the way we are using Dask!
+
+  ## Use cases
+
+  First, let's have a look at the use cases we have for Dask. The use of Dask at Blue Yonder
+  is strongly coupled to the concept of [Datasets](https://tech.jda.com/introducing-kartothek/),
+  tabular data stored in blob storage. We use these datasets for storing everything that is needed
+  for generating predictions, recommended order quantities, and recommended prices for our customers,
+  as well as the predictions, order quantities, and prices themselves.
+
+  ### Downloading data from a database to a dataset
+
+  One typical use case that we have is downloading large amounts of data from a database into a
+  dataset. We use Dask/Distributed here as a means to parallelize the download and the graph is
+  an extremely simple: a lot of parallel nodes downloading pieces of data into a blob and one
+  reduction node writing the dataset metadata (which is basically a dictionary what data can be found in which blob).
+
+  ### Doing computations on a dataset
+
+  A lot of our algorithms for machine learning, forecasting, and optimization work on datasets.
+  A source dataset is read, computations are performed on the data, and the result is written to
+  a target dataset.
+  In many cases, the layout of the source dataset (the partitioning, i.e., what data resides in which blob)
+  is used for parallelization. This means the algorithms work independently on the individual
+  blobs of the source dataset. Therefore our respective Dask graphs again look very simple, with parallel nodes each performing
+  the sequential operations of reading in the data from a source dataset blob, doing some computation on it,
+  and writing it out to a target dataset blob. Again, there is a final reduction node writing the target
+  dataset metadata.
+
+  ### Dataset maintenance
+
+  We also use Dask/Distributed for performing maintenance operations like garbage collection on datasets.
```

Comment:
Given that we are already separating the two points by using headers (which are `Dataset Maintenance and `Doing computations on a dataset`), would it make sense to use `we also`?

Reply:
Was probably from a time before I introduced the headers; removed the also.